src/App.js

```diff
@@ -28,25 +31,91 @@ const generateSquares = () => {
 const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
-  const [squares, setSquares] = useState(generateSquares());
+  const [squares, setSquares] = useState(generateSquares()); // useState calls generateSquares() to sets the initial state of squares
+  const [player, playerTurn] = useState(PLAYER_1); // sets initial state to PLAYER_1 > the first move will always be 'x'
+  let [winner, checkWinner] = useState(''); 
+
+  // function to update state to board
+  const onClickCallback = (id) => { // when a square component is clicked, its id prop value is passed to this function
+    
+  if (winner) { // this logic makes the squares unclickable once we have a winner
+    return;
+  }
+
+    const newSquares = squares.map((squareArray) => { // this logic creates a new object, which is ultimately used to re-render the board with new states
+      for (const insideSquare of squareArray) {
+        if (insideSquare.id === id) { // finds the square whose id matches the id that is passed in when onClickCallback is called
+          insideSquare.value = player; // and then sets that square's value to either 'x' or 'o', depending on the next part of the function
+        }
+      }
+      return squareArray;
+    });
 
-  // Wave 2
-  // You will need to create a method to change the square
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+    if (player === PLAYER_1) { // logic to toggle which player's turn it is, which changes whether to set the square's value as 'x' or 'o'
```

Comment:
Related to the above bug, the player should only change if the current player has made a valid move.